### DISCUSSION TITLE
UAC2 192kHz sampling rate

### DISCUSSION THREAD

**Discussion by Dodotronic** [hathach/tinyusb#2073] [General]
Hi all,
it seems that when I change the sampling rate:
#define AUDIO_SAMPLE_RATE   192000
in UAC2 4 channel example the device is not recognized
The example works well up to 96000 but I need higher sampling frequencies
I don't know how to proceed with the debug
Any idea?
Thanks 
Iv

**Comment by HiFiPhile** [comment]
Hi, what's your hardware ?

You need USB High-Speed for 24/192.

**Reply by Dodotronic** [reply]
yes I have Hi Speed device it's a STM32H7B3I-DK
It seems that the descriptor is not valid
here the debug:
USBD init on controller 1
sizeof(usbd_device_t) = 55
sizeof(tu_fifo_t) = 12
sizeof(tu_edpt_stream_t) = 24
AUDIO init
USBD Bus Reset : Full Speed

USBD Setup Received 80 06 00 01 00 00 40 00
  Get Descriptor Device
  Queue EP 80 with 18 bytes ...
USBD Xfer Complete on EP 80 with 18 bytes
  Queue EP 00 with 0 bytes ...
USBD Xfer Complete on EP 00 with 0 bytes
USBD Bus Reset : Full Speed

USBD Setup Received 00 05 0F 00 00 00 00 00
  Set Address
USBD Xfer Complete on EP 80 with 0 bytes

USBD Setup Received 80 06 00 01 00 00 12 00
  Get Descriptor Device
  Queue EP 80 with 18 bytes ...
USBD Xfer Complete on EP 80 with 18 bytes
  Queue EP 00 with 0 bytes ...
USBD Xfer Complete on EP 00 with 0 bytes

USBD Setup Received 80 06 00 02 00 00 FF 00
  Get Descriptor Configuration[0]
  Queue EP 80 with 64 bytes ...
USBD Bus Reset : Full Speed

USBD Setup Received 80 06 00 01 00 00 40 00
  Get Descriptor Device
  Queue EP 80 with 18 bytes ...
USBD Xfer Complete on EP 80 with 64 bytes
  Queue EP 00 with 0 bytes ...
USBD Xfer Complete on EP 00 with 0 bytes
USBD Bus Reset : Full Speed

USBD Setup Received 80 06 00 01 00 00 40 00
  Get Descriptor Device
  Queue EP 80 with 18 bytes ...
USBD Xfer Complete on EP 80 with 18 bytes
  Queue EP 00 with 0 bytes ...
USBD Xfer Complete on EP 00 with 0 bytes
USBD Bus Reset : Full Speed

USBD Setup Received 80 06 00 01 00 00 40 00
  Get Descriptor Device
  Queue EP 80 with 18 bytes ...
USBD Xfer Complete on EP 80 with 18 bytes
  Queue EP 00 with 0 bytes ...
USBD Xfer Complete on EP 00 with 0 bytes
USBD Suspend : Remote Wakeup = 0

**Comment by Dodotronic** [comment]
Thanks HiFiPhile!
just solved 
Here the solution
when define:
#define BOARD_TUD_MAX_SPEED OPT_MODE_DEFAULT_SPEED
the device doesn't set the usb to the High speed as it should 
you need to specifically define:
#define BOARD_TUD_MAX_SPEED   OPT_MODE_HIGH_SPEED

**Comment by Dodotronic** [comment]
at the moment the device is recognized but when recording nothing happens

**Comment by Dodotronic** [comment]
now I'm trying with 384kHz but with no success, the device is not recognized
